plugins/modules/managed_serviceaccount_addon.py

```diff
@@ -347,7 +274,13 @@ def execute_module(module: AnsibleModule):
     # get token
     token_bytes = base64.b64decode(secret.data.token)
     token = token_bytes.decode('ascii')
-    module.exit_json(token=token)
+    managed_serviceaccount = {
+        'name': managed_service_account.metadata.name, 'namespace': managed_service_account.metadata.namespace,
+        'api_version': managed_service_account.apiVersion, 'kind': managed_service_account.kind,
+        'uid': managed_service_account.metadata.uid, 'service_account': {'name': managed_service_account.metadata.name,
+                                                                         'namespace': managed_service_account_addon.spec.installNamespace, 'token': token}
```

Comment:
we can probably simplify the output a little to scope to the information we need to expose to the user
```             
{
  "name": "name of the ManagedServiceAccount resource",
  "namespace": "namespace of the ManagedServiceAccount resource", //cluster-namespace
  "token": "token of the ServiceAccount"
}
```
we need to examine how do we do deletion of managed-serviceaccount in "ansible style"